cmd/tegola/cmd/root.go

```diff
@@ -75,7 +78,8 @@ func initConfig(configFile string, cacheRequired bool) (err error) {
 	}
 
 	// init our maps
-	if err = register.Maps(nil, conf.Maps, providers); err != nil {
+	// note that we are sending the whole config file to include both maps and providers
+	if err = register.Maps(nil, conf, providers); err != nil {
```

Comment:
I don't think you need to send the whole map in. I think we can still send in just `conf.Maps`

Reply:
You'll have to help me with this. I couldn't figure out another way to access the "auto" or "auto_config" on the conf.Providers.

Reply:
What if conf.Maps was generated and then passed in?

Reply:
Yea. That may be necessary. I still haven't sorted through the whole "No Maps Defined" scenario yet